doc/reference/trd-legal.md

```diff
@@ -0,0 +1,178 @@
+http://opensource.org/licenses/MIT).
+
+All contributions to the Tock kernel (the code hosted at
+https://github.com/tock/tock) MUST be licensed under these terms.
+
+
+3 Copyright
+===========
+
+Entities contributing resources to open-source projects often require
+attribution to recognize their efforts. Copyright notices are a common means to
+provide this. For downstream users, the license terms of Tock ensures
+unencumbered use.
+
+Copyrights in Tock projects are retained by their contributors. No
+copyright assignment is required to contribute to Tock projects.
+
+Artifacts in the Tock project MAY include explicit copyright notices.
+Substantial updates to an artifact MAY add additional copyright notices to an
+artifact. In general, modifications to a file are expected to retain existing
+copyright notices.
+
+For full authorship information, see the version control history.
+
+
+4 Implementation
+================
+
+All textual files that allow comments MUST include a license notice and
+copyright notice(s).
+
+Copyright notices SHOULD include a year. Newer copyright notices SHOULD be
+placed after existing copyright notices. If non-trivial updates are performed by
+an original copyright author, they MAY amend the year(s) indicated on their
+existing copyright statement or MAY add an additional copyright line, at their
+discretion.
+
+
+4.1 Format
+----------
+
+License and copyright information SHOULD have exactly one (1) blank line
+separating it from any other content in the file.
+
+Text described in this section SHOULD be pre-fixed or post-fixed with
+technically necessary characters (i.e. to mark as a comment in source
+code) as appropriate.
+
+The first line of license text SHOULD appear as-follows:
+
+> Licensed under the Apache License, Version 2.0 or the MIT License.
+
+The second line of license SHOULD adhere to the [SPDX](https//spdx.dev)
+specification for license description. As of this writing, it SHOULD
+appear as-follows:
+
+> SPDX-License-Identifier: Apache-2.0 OR MIT
+
+The [current (v2.3) normative
+rules](https://spdx.github.io/spdx-spec/v2.3/SPDX-license-expressions/) permit
+case-insensitive matches of the license identifier but require case-sensitive
+matching of the disjunction operator. To simplify enforcement of licensing and
+documentation rules, license information SHOULD preserve case as-shown in the
+SPDX license list (i.e. as-presented above).
+
+Copyright lines SHOULD follow this pattern:
+
+> Copyright {entity} {year}([,year],[-year]).
+
+The `{entity}` field should reflect the entity wishing to claim copyright. The
+`{year}` field SHOULD reflect when the copyright is first established.
+Substantial updates in the future MAY indicate renewed copyright, via additional
+comma-separated years or via range syntax, at the copyright holder’s discretion.
+The initial year SHALL NOT be removed unless it is the express intent of the
+copyright holder to relinquish the initial copyright.
+
+
+### 4.1.1 Examples
```

Comment:
The Examples section needs to be updated to indicate that a copyright line is necessary.

We seem to have come to the conclusion (see [Hudson's comment](https://github.com/tock/tock/pull/3318#issuecomment-1324225961)) that we should require "Copyright Tock Contributors" to be one of the copyright lines, or at least make it standard, so it should probably be part of the examples as well.

Reply:
At the risk of repeating what I said above, I’d vouch for making it standard, as I foresee issues with requiring this line for files added without modification by a “Tock Contributor”. Otherwise I agree that this section should be updated.

Reply:
Merged #3377 which updated the examples. `Copyright Tock Contributors` is part of the examples but is not required in the text.